### DISCUSSION TITLE
侧边栏同时展示多个标签页

### DISCUSSION THREAD

**Discussion by linonetwo** [tiddly-gittly/tiddlywiki-plugins#33] [Ideas]
<img width="1245" alt="8F3BF21F36AAB266F4416D5A31E9A7BE" src="https://user-images.githubusercontent.com/3746270/150278548-98e615e4-3e70-4a21-b8b4-778f94c3aa08.png">

Simultaneously show multiple tabs on sidebar vertically (VSCode style), drag to resize height. 在侧边栏同时展示多个垂直的标签（VSCode风格），拖动可以调整高度。

我优先做这个插件是为了实现昨天设想的这个提示功能，在打开某条笔记时，自动展示一些相关提示。
原来只要搞一个新的 $:/core/ui/SideBarSegments/tabs 覆盖原版的，就能做出 vscode 风格的侧边栏了

![913BF06F1118C63669413788C5B3F1EB](https://user-images.githubusercontent.com/3746270/150278686-ab5df895-af93-4040-b4a4-b929b43bc84a.png)

我的需求就是能同时一半展示文件目录，一半展示「最近」或者「脑图」，这么多 tab

<img width="438" alt="9BBC242658D784AEBE28F187D9D7C484" src="https://user-images.githubusercontent.com/3746270/150278638-7d8ba186-641c-4523-a4f3-d69c4014d3d3.png">

<img width="457" alt="AECE764413AB61831612D1412ADBC2A0" src="https://user-images.githubusercontent.com/3746270/150278631-efd88af6-4a3e-41b6-bab4-e62efa74d0f7.png">

不过跟 tw 的结构不大一样， vscode 侧边栏有两级，tw 只有一级

希望像 VSCode 那样可以拖动每个分隔的大小
<img width="484" alt="FA0F1F2491B5E849E69DD69E57D179FC" src="https://user-images.githubusercontent.com/3746270/150278740-47e0780c-75a6-47b6-86a2-21d4c079b5ca.png">

我用 wikitext 是写不出这么复杂的结构了，还是用 JS 写吧

![8D28DD3CC87D8B11CD0F0C65E8AD2A2D](https://user-images.githubusercontent.com/3746270/150278590-dd81f568-a98e-4eda-a05a-d8a4350960a8.jpg)
基本就是这个效果了，它估计也是改了 $:/core/ui/SideBarSegments/tabs 吧

可以这样设计

<img width="437" alt="8B6A8445F31D1D76FD597F6A3ACA26C2" src="https://user-images.githubusercontent.com/3746270/150278761-4f8685a7-562e-427a-8c11-18ae260d2065.png">

**Comment by linonetwo** [comment]
如何解决想看一个新tab的问题？例如点开之前关着的「最近」，应该就是希望把它打开大一点儿然后使用里面的功能，那么这时候应该要给新打开的标签至少一半的大小，再次点击这个tab之后还原之前已有tab的大小。

**Reply by linonetwo** [reply]
不过vscode的好像有更复杂的规则，二期可以搞看看，不过也不一定就适合 tw 的场景。
它可以通过右键开关各个tab，不过tw里面顶部的 tab 按钮列表可能得搞一个单独的可伸缩tab来管理。

**Comment by dongrentianyu** [comment]
其实换一个思路，我们可以直接转换正文模板里的内容。也就是点一下按钮，加上某个标签，这样就展示正文内容，再点一下，就把之前的按钮替换成另一个按钮，然后就只展示脑图或者最近打开的页面，或者当前条目的目录内容。再点一下，又可以转换成其他内容了。如果要转换的很多就可以在按钮上搞一个下拉框来选择。我正在尝试实现这样的功能。理论上应该就和快速加入私有笔记的按钮是一样的。

**Comment by dongrentianyu** [comment]
如果是多个的话，可以下拉也做成按钮的形式。这样就不会用文字了，显得很难看了。这个应该也可以抄一抄默认模板里的页面下拉按钮设计。
